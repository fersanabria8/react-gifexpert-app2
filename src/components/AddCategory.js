import React, { useState } from 'react'
import PropTypes from "prop-types";


export const AddCategory = ({setCategories}) => {

  const [inputValue, setInputValue ] = useState(''); 

  const handleInputChange = (e) => {
    setInputValue ( e.target.value );

    // console.log('handleInputChange llamado') // este lo dejo para hacer pruebas test de que funciona al cambiar. ( e.target.value ) = en el test vemos como llamar a los elementos e.targe.value, porque sino nos va a tirar error

  }

  const handleSubmit = (e) => {
      e.preventDefault();

      console.log('handleSubmit', inputValue);

      if( inputValue.trim().length > 2){
        setCategories( cats => [ inputValue, ...cats]);
        setInputValue('');
      }
  }

  
  return (
    <form onSubmit={ handleSubmit }>
      <p> { inputValue } </p>
      <input 
          type="text" 
          value={ inputValue }
          onChange={ handleInputChange }
      /> 
    </form>
  )
}
AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired
}
