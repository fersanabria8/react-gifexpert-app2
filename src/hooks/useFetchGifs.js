import { useEffect, useState } from "react"
import { getGifs } from "../helpers/getGifs";

export const useFetchGifs = ( category ) => {
  

  // INICIALIZAMOS CUSTOM HOOK 
  const [state, setState] = useState({
    data: [],
    loading: true
  });

  // DISPARAMOS UN EFECTO useEffect
  useEffect( () => {

    // ESPERAMOS QUE SE DISPARE ALGUNA TAREA ASINCRONA
    getGifs( category )
      .then( imgs => {
        // PARA CAMBIAR NUEVAMENTE EL ESTADO Y CUANDO CAMBIA LOS ESTADOS... HACEMOS NUESTRAS PRUEBAS TEST
          setState({
            data: imgs,
            loading: false
          });
      })
  },[ category ])

  return state; // {data:[], loading: true};
}