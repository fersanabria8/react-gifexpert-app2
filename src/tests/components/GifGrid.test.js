import React from 'react';
import { shallow } from "enzyme";
import { GifGrid } from "../../components/GifGrid";
import '@testing-library/jest-dom';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');



describe('Pruebas en el <GifGrid />', () => {

  const category = 'One Punch'

  test('Debe de mostrar el componente correctamente', () => {

    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });

    const wrapper = shallow( <GifGrid category= { category } /> )// ahora esta undefined
    expect( wrapper ).toMatchSnapshot();
  });

  test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {

    const gifs = [{
      id: 'ABC',
      url: 'http://localhost/asdasdsad.jpg',
      title: 'Cualquier cosa'
    },
    
    {
      id: 'A123',
      url: 'http://localhost/asdasdsad.jpg',
      title: 'Cualquier cosa'
    }]

    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });

    const wrapper = shallow( <GifGrid category={ category } /> );

    // expect( wrapper ).toMatchSnapshot();
    expect( wrapper.find('p').exists() ).toBe(false);
    expect( wrapper.find('GifGridItem').length ).toBe ( gifs.length );
  })


});