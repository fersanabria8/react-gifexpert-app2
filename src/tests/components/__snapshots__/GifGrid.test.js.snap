// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el <GifGrid /> Debe de mostrar el componente correctamente 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`Pruebas en el <GifGrid /> Debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
     
    One Punch
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="http://localhost/asdasdsad.jpg"
    />
    <GifGridItem
      id="A123"
      key="A123"
      title="Cualquier cosa"
      url="http://localhost/asdasdsad.jpg"
    />
  </div>
</Fragment>
`;
